### PR TITLE
"undefined is not an object" in `promise.finally` of promiseThrottler

### PR DESCRIPTION
**Pull request by vogdb**

### CODE REVIEW

**Comment by vogdb** [comment]
@njam please review. Unfortunately can't reproduce, so have to monkey-patch here.

**Comment by njam** [comment]
@zazabe could you review?

**Comment by zazabe** [comment]
I can imagine it happens because [some throttled promises](https://github.com/cargomedia/sk/blob/master/library/SK/library/SK/Form/SearchContent.js#L32) use [`cm.router.route`](https://github.com/cargomedia/cm/blob/master/library/CM/App.js#L1435), which may return `null` in some conditions (bug introduced by me :/ )...

Anyhow, what about wrapping `fn.apply(this, arguments)` by a `Promise.try()` to always handle its return as a promise?

**Comment by vogdb** [comment]
> I can imagine it happens because [some throttled promises](https://github.com/cargomedia/sk/blob/master/library/SK/library/SK/Form/SearchContent.js#L32) use [`cm.router.route`](https://github.com/cargomedia/cm/blob/master/library/CM/App.js#L1435), which may return `null` in some conditions (bug introduced by me :/ )...

Thank you so much for this. I've missed this. Thought we always return promises in out usages. Will try to fix it first and see

**Comment by vogdb** [comment]
@zazabe please look

**Comment by zazabe** [comment]
looks good, but it would be safer to also check if the `promiseThrottler` function returns a `Promise`, no? 
wrapping with `Promise.try()` could be tricky because it will postpone its execution, but maybe the same code as before would be enough...

**Comment by vogdb** [comment]
> but maybe the same code as before would be enough...

@zazabe you mean @1a541b4 ? Should I add `instanceof Promise`?

File: client-vendor/after-body/promise-utils/promise-throttler.js [hunk 1]
```diff
@@ -48,10 +48,13 @@
     }
 
     return function() {
-      if (!promise || !promise.isPending()) {
+      if (!promise) {
         promise = fn.apply(this, arguments);
-        promise.finally(resetClosurePromise.bind(null, promise));
-        return promise;
+        //we check here for `promise` again cause it can be nullified `resetClosurePromise`
```
Comment (zazabe) [inline client-vendor/after-body/promise-utils/promise-throttler.js]: it's more about what `fn` returns I guess

**Comment by vogdb** [comment]
@zazabe please review

File: library/CM/App.js [hunk 2]
```diff
@@ -1433,7 +1433,7 @@ var CM_App = CM_Class_Abstract.extend({
       var urlBase = cm.options.urlBase;
       if (0 === url.indexOf(urlSite)) {
         var path = url.substr(urlBase.length);
-        cm.getDocument().loadPage(path);
+        return cm.getDocument().loadPage(path);
```
Comment (zazabe) [inline library/CM/App.js]: hope it's the only misusage of `promiseThrottler`...

**Comment by zazabe** [comment]
Oops, `qunit` tests are broken, could you open a followup @vogdb?

**Pull request merged**